File: WINGS_AI_Bootcamp/Haskell/Rot13_Encoder.hs
import Data.Char (chr, ord, isLower, isUpper)

rot13Char :: Char -> Char
rot13Char c
  | isLower c = chr (((ord c - ord 'a' + 13) `mod` 26) + ord 'a')
  | isUpper c = chr (((ord c - ord 'A' + 13) `mod` 26) + ord 'A')
  | otherwise = c

rot13 :: String -> String
rot13 = map rot13Char

main :: IO ()
main = putStrLn (rot13 "Hello World!")

File: WINGS_AI_Bootcamp/Elixir/Rot13_Encoder.ex
defmodule Rot13 do
  def encode(text) do
    String.to_charlist(text)
    |> Enum.map(&rot13_char/1)
    |> List.to_string()
  end

  defp rot13_char(c) when c in ?a..?z do
    rem(c - ?a + 13, 26) + ?a
  end

  defp rot13_char(c) when c in ?A..?Z do
    rem(c - ?A + 13, 26) + ?A
  end

  defp rot13_char(c), do: c
end

IO.puts Rot13.encode("Hello World!")

